bump(x11/audacity): 3.6.4

twaik opened the pull request:
Fixes #18790
Fixes #10412
Fixes #21545
Fixes #20570
Fixes #19679
Fixes #5422

Audacity starts, can open wav and mp3 files, play and record sound (with RECORD_AUDIO permission granted to Termux:API). Some review needed.

Biswa96 commented:
I have tested the aarch64 CI artifact in my phone but got an error.

```
CANNOT LINK EXECUTABLE "audacity": library "lib-theme-resources.so" not found: needed by main executable
```

After exporting the variables like this file https://github.com/audacity/audacity/blob/master/au3/linux/audacity.sh

```
$ LD_LIBRARY_PATH=$PREFIX/lib/audacity AUDACITY_MODULES_PATH=$PREFIX/lib/audacity/modules audacity
Bad system call
```
 audacity fails with semget syscall.

twaik commented:
Ok, so I removed usage of semget/semctl/semop and also shmget and managed to make it work without them. Also I added `-Wl,-rpath` to make it work without LD_LIBRARY_PATH. Should work now.

twaik commented:
By `should work` I meant the patch worked fine with audacity built for regular linux, I did not check it on my own device.

Biswa96 commented:
Now audacity opens does not play any audio file. Try to import a mp3 file and click on the play button.

twaik commented:
I tried to open m4a file and it refused to open it too. Probably it can not pick ffmpeg.

twaik commented:
Interesting. The same happens to old audacity, probably portaudio related.

twaik commented:
[portaudio.tar.gz](https://github.com/user-attachments/files/17401084/portaudio.tar.gz)
Fun. I managed to make portaudio with opensles using some old sources, so output works, but no input.
And this portaudio crashes after a finishing playing my sample sound, probably some bug in portaudio or in opensles hostapi backend.

twaik commented:
Sample to enumerate portaudio devices.
```
#include <stdio.h>
#include "portaudio.h"

int main(void)
{
    PaError err;

    /* Initialize PortAudio */
    err = Pa_Initialize();
    if (err != paNoError) {
        fprintf(stderr, "PortAudio error during initialization: %s\n", Pa_GetErrorText(err));
        return 1;
    }

    /* Get the number of audio devices */
    int numDevices = Pa_GetDeviceCount();
    if (numDevices < 0) {
        fprintf(stderr, "PortAudio error getting device count: %s\n", Pa_GetErrorText(numDevices));
        Pa_Terminate();
        return 1;
    }

    printf("Number of devices found: %d\n\n", numDevices);

    /* Iterate over each device */
    for (int i = 0; i < numDevices; i++) {
        const PaDeviceInfo *deviceInfo = Pa_GetDeviceInfo(i);

        printf("Device #%d:\n", i);
        printf("  Name                        : %s\n", deviceInfo->name);
        printf("  Host API                    : %s\n", Pa_GetHostApiInfo(deviceInfo->hostApi)->name);
        printf("  Max input channels          : %d\n", deviceInfo->maxInputChannels);
        printf("  Max output channels         : %d\n", deviceInfo->maxOutputChannels);
        printf("  Default sample rate         : %.2f\n", deviceInfo->defaultSampleRate);
        printf("  Default low input latency   : %.4f\n", deviceInfo->defaultLowInputLatency);
        printf("  Default low output latency  : %.4f\n", deviceInfo->defaultLowOutputLatency);
        printf("  Default high input latency  : %.4f\n", deviceInfo->defaultHighInputLatency);
        printf("  Default high output latency : %.4f\n", deviceInfo->defaultHighOutputLatency);
        printf("\n");
    }

    /* Terminate PortAudio */
    err = Pa_Terminate();
    if (err != paNoError) {
        fprintf(stderr, "PortAudio error during termination: %s\n", Pa_GetErrorText(err));
        return 1;
    }

    return 0;
}
```

twaik commented:
Ok, I figured out. For some reason `pthread_join` crashed application instead returning immediately or throwing `ESRCH` error. Now sound output works.

twaik commented:
AFAIK microphone support is not possible without RECORD_AUDIO so it will be `wontfix` until @agnostic-apollo decides to add it to termux-app's application manifest.

agnostic-apollo commented:
You can install termux-api, will have to see about adding to termux-app itself.

https://github.com/termux/termux-api/blob/c553bc2ba7de37ad5c42df23d71e699d2785d7c5/app/src/main/AndroidManifest.xml#L27

twaik commented:
Wow! So microphone works fine too! So cool!
I did not know termux:api can give permissions to tools of main termux app.

agnostic-apollo commented:
They share  `sharedUserId`, so permissions are shared between all apps under the same uid regardless of who requests it.

twaik commented:
I'll merge this PR in 24 hours if no one minds.

agnostic-apollo commented:
Oh, so you finally have the audacity to merge! huh!

truboxl commented:
Should turn off networking, update checks and crash reports

truboxl commented on x11-packages/audacity/build.sh:
```diff
@@ -2,47 +2,116 @@ TERMUX_PKG_HOMEPAGE=https://www.audacityteam.org/
+-Daudacity_has_tests=no
+-Daudacity_has_networking=no
+-Daudacity_has_crashreports=no
+-Daudacity_has_sentry_reporting=no
```
Missing `-Daudacity_has_updates_check=off`

twaik replied:
Done.

twaik commented:
Fixed a sqlite-related problem. 
```
 $ audacity
 CANNOT LINK EXECUTABLE "audacity": cannot locate symbol "sqlite3_trace" referenced by "/system/lib64/libandroid_runtime.so"...
```
